packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -472,43 +471,71 @@ export class ConnectionService {
       const recipientKey = messageContext.recipientKey && messageContext.recipientKey.publicKeyBase58
       const senderKey = messageContext.senderKey && messageContext.senderKey.publicKeyBase58
 
-      if (previousSentMessage) {
-        // If we have previously sent a message, it is not allowed to receive an OOB/unpacked message
-        if (!recipientKey) {
-          throw new AriesFrameworkError(
-            'Cannot verify service without recipientKey on incoming message (received unpacked message)'
-          )
-        }
+      // set theirService to the value of lastReceivedMessage.service
+      let theirService =
+        messageContext.message?.service?.resolvedDidCommService ??
+        previousReceivedMessage?.service?.resolvedDidCommService
```

Comment:
nit: Before it was referred to `lastReceivedMessage` might be good to name them consistently